### ISSUE TITLE
2.4.6 complicated by works of fiction

### ISSUE THREAD

**Issue by mattgarrish** [w3c/wcag#2039]
Having headings that describe the purpose of each section of a non-fiction work is effectively a requirement in any reasonably well-written work, but how is this success criterion supposed to apply to works of fiction?

The chapter headings in a novel don't necessarily clearly describe a topic or purpose, for example. So do novels with esoteric headings fail 2.4.6?

The success criterion has a very functional bias to it, at least as I read it. There doesn't appear to be any wiggle room to handle content that isn't strictly structured by topic.

**Comment by scottaohara** [comment]
i would submit in this use case the chapter title is not necessarily meant to be a summation of the content that follows, but merely indicate "this is the name of the chapter you are about to read", and by naming the chapter you have thus provided a useful title/heading to describe it.   i'd be very surprised to see someone reporting on this in the context you presented.

**Comment by patrickhlauke** [comment]
the whole "how to determine what is and isn't descriptive enough" part is completely wooly and subjective, so there's plenty of wiggle room i'd say. and yes for works of fiction, you can basically pass and handwave it. doubt anybody would then come back and say "nah-ah! you fail 2.4.6!", and if they do, let a judge/lawyer work it out...

**Comment by patrickhlauke** [comment]
@scottaohara partial "snap"

**Comment by mattgarrish** [comment]
> i would submit in this use case the chapter title is not necessarily meant to be a summation of the content that follows

Sure, that's essentially my point. But the wording of the success criterion doesn't account for this. It doesn't say the heading has to describe the purpose or topic only if there is a specific purpose or topic, or that it can just generally label the section. It says that headings have to describe the purpose or topic.

> but merely indicate "this is the name of the chapter you are about to read", and by naming the chapter you have thus provided a useful title/heading to describe it.

Sure, but that's the kind of handwaving that @patrickhlauke suggests. It's why I've always ignored this success criterion for publishing, but it's troubling to say to people to ignore a success criterion because it doesn't make a lot of sense outside a specific context. It would be much better if the success criterion were clearer about its application.

It's also problematic because the part about labels describing the purpose of inputs, for example, is useful, so you can't completely ignore it.

**Comment by bruce-usab** [comment]
> ...the wording of the success criterion doesn't account for this.

I respectfully disagree.  My recollection is that the context of fiction book chapter titles was one of the use cases considered with the normative phrasing chosen for SC 2.4.6.

Please see [Understanding](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html), emphasis added.

> The intent of this Success Criterion is to help users understand what information is contained in Web pages and how that information is organized. When headings are clear and descriptive, users can find the information they seek more easily, and they can understand the relationships between different parts of the content more easily. Descriptive labels help users identify specific components within the content.

> Labels and headings do not need to be lengthy. **A word, or even a single character, may suffice if it provides an appropriate cue to finding and navigating content**.

If a word may suffice, then a cryptic few words certainly is enough when, as is the usually case with works of fiction, the few words chosen for the chapter title are deliberately chosen.

I do wish that some of the limited implications that follows from the intent behind this SC was better conveyed through a plain reading of the SC phrasing.

@mattgarrish if you can suggest a change to the SC phrasing, which does not change its current meaning, I would (personally) appreciate seeing that.

Circling back your OP in this issue:

> The chapter headings in a novel don't necessarily clearly describe a topic or purpose, for example.  So do novels with esoteric headings fail 2.4.6?

No.  Because the requirement is merely <q>describe topic or purpose</q>.  The esoteric chapter headings found in novels (1) are descriptive, and (2) well suited to their purpose.

> The success criterion has a very functional bias to it, at least as I read it.

Agreed, and the esoteric chapter headings found in novels are very functional.

> There doesn't appear to be any wiggle room to handle content that isn't strictly structured by topic.

I don't disagree that a plain reading has this implication.   It is a defect, IMHO, that one has to resort to the Understanding to find this sort of nuance.

**Comment by patrickhlauke** [comment]
> Agreed, and the esoteric chapter headings found in novels are very functional.

the tension is that they can be non-descriptive (on purpose), even saying the opposite of what the topic/content of the chapter actually is, and that seems to contravene the normative "Headings and labels *describe* topic or purpose". e.g. if an author uses a chapter or section title "A peaceful morning" ironically while the text that follows describes some atrocities in a war-torn country, does it "describe topic or purpose"?

As @bruce-usab points out, the understanding then seems to pivot away from "descriptive" to just "providing some cue/landmark" ("A word, or even a single character, may suffice if it provides an appropriate cue to finding and navigating content"). Sadly that isn't reflected in the normative wording.

**Comment by bruce-usab** [comment]
I agree that there is tension.  There is not a requirement that headings (or labels) _**be**_ descriptive.  (Although, I agree, that is a reasonable inference.)  Headings (and labels) are required to describe topic _or_ to describe purpose.  The text of a (fiction/novel) chapter title _does_ describe the topic!  I love the example @patrickhlauke gives of <q>A peaceful morning</q> because it points out just how very much that chapter title (heading) memorably identifies (describes) the main substance (topic) of the chapter.

@mattgarrish wrote:
> It's why I've always ignored this success criterion for publishing, but it's troubling to say to people to ignore a success criterion because it doesn't make a lot of sense outside a specific context.

That is not how I would characterize the reality.  In the context of publishing, this SC is always satisfied.  You can ignore this SC for publishing because it is probably impossible to find a real-world failure.  This contrasts to the web where there have been examples of pages with headings that seem entirely random.

And just because a single character _might_ suffice (maybe for legislation or regulatory document) it does not follow that this is enough most of the time.



It would be much better if the success criterion were clearer about its application.

**Comment by mattgarrish** [comment]
> @mattgarrish if you can suggest a change to the SC phrasing, which does not change its current meaning, I would (personally) appreciate seeing that.

Ya, this is hard as my understanding of describing a topic diverges from the success criterion's so I'm not sure what to do that would keep the useful aspects of it while also expanding the scope. I'll have to think about it some more.

What's there now is a fine description for many cases, don't get me wrong. It's just the suggestion in it that headings that diverge from a strict description of what events will be covered in a chapter, and that those events have to have a neat topic to them, are invalid.

The examples in the understanding document reinforce this by highlighting content that can be easily broken down by topic.

A chapter heading, however, might set a mood or theme, or simply introduce some levity or absurdity for the content that follows.

**Comment by mattgarrish** [comment]
The best thought I've had so far would be to avoid specific terms like "topic" and "purpose" in the SC text itself and instead focus on the context aspect, so:

> Headings and labels establish context

Context doesn't require a topic or purpose but seems to be the underlying need here. If I use chapter numbers, for example, these don't tell you anything about the topic or purpose of the content, but they do establish context (i.e., how far into a book you are). Whatever the meaning of a written heading, it again provides some kind of context to what follows.

Now whether this works in place of the current wording I don't know. Maybe it loosens things up too much for the web heading problem mentioned earlier. Maybe it could be combined into the SC to make "Headings and labels describe topic or purpose or similarly establish context".

**Comment by JAWS-test** [comment]
I don't think it makes sense to rewrite the SC, because then there is a weakening of the requirement with respect to labels. Labels in forms should be meaningful (and e.g. not ironic or consisting of only one character), for headings this is indeed less important. This can be taken into account in WCAG 3.0 by considering labels and headings separately. Now I would be more in favor of an addition in Understanding that headings are not subject to such high requirements under certain circumstances (works of fiction).

**Comment by bruce-usab** [comment]
@mattgarrish I agree with @JAWS-test that your proposed phrasing weakens the requirement with respect to labels.  It would also introduce the new term/concept about what it means to <q>establish context</q>.  Which is probably a valuable discussion, but I am hoping that we might hit upon something that, theoretically, could be addressed as an erratum.  Borrowing from 1.1.1 what do people think of this: 

> Headings and labels provide descriptive identification of the topic or purpose.
